climakitae/selectors.py

```diff
@@ -303,64 +358,149 @@ class DataSelector(param.Parameterized):
     UI could in principle be used to update these parameters instead.
     """
 
-    choices = param.Dict(dict())
+    # Defaults 
     default_variable = "Air Temperature at 2m"
-    variable = param.ObjectSelector(default=default_variable, objects=dict())
-    timescale = param.ObjectSelector(
-        default="monthly", objects=["hourly", "daily", "monthly"]
-    )  # for WRF, will just coarsen data to start
+    default_scenario = ["Historical Climate"]
     time_slice = param.Range(default=(1980, 2015), bounds=(1950, 2100))
-    scenario = param.ListSelector(objects=dict())
-    resolution = param.ObjectSelector(objects=dict())
     append_historical = param.Boolean(default=False)
-    descrip_dict = _read_var_csv(CSV_FILE, index_col="description")
-    variable_description = param.String(
-        default=descrip_dict[default_variable]["extended_description"],
-        doc="Extended description of variable selected"
-    )
+    area_average = param.Boolean(default=False)
 
+    resolution = param.ObjectSelector(
+        default="45km", 
+        objects=["45km","9km","3km"]
+    ) 
+    timescale = param.ObjectSelector( 
+        default="monthly", 
+        objects=["hourly","daily", "monthly"]
+    )
+        
+    # Empty params, initialized in __init__ 
+    dataset = param.ObjectSelector(objects=dict())
+    scenario = param.ListSelector(objects=dict())
+    variable = param.ObjectSelector(objects=dict())
     units = param.ObjectSelector(objects=dict())
+    extended_description = param.ObjectSelector(objects=dict())
+    variable_id = param.ObjectSelector(objects=dict())
 
     def __init__(self, **params):
         # Set default values 
         super().__init__(**params)
-        self.param["resolution"].objects = self.choices["resolutions"]
-        self.resolution = self.choices["resolutions"][0]
-        _list_of_scenarios = list(self.choices["scenarios"]["45 km"].keys())
-        self.param["scenario"].objects = _list_of_scenarios
-        self.scenario = ["Historical Climate"]
-        self.param["variable"].objects = self.choices["variable_choices"]["hourly"][
-            "Dynamical"
-        ]
-        self.units = self.descrip_dict[self.variable]["native_unit"]
+        
+        # Dataset selection 
+        self.dataset = "WRF"
```

Comment:
.dataset is not a good name for this, given that it is also the word used for an xarray Dataset... can we call it 'product'? or 'downscaling_method'?